File: 258-AddDigits/258-AddDigits.py
# Last updated: 7/8/2026, 4:48:22 PM
class Solution(object):
    z = 0
    def di(self,a):
        if a%10 == a:
            return a
        z = 0
        while a > 0:
            z = z+a%10
            a = a//10
        return self.di(z)
    def addDigits(self, num):
        return self.di(num)

File: 258-AddDigits/258-AddDigits.java
// Last updated: 7/8/2026, 4:48:26 PM
class Solution {
    public int addDigits(int num) {
        if(num<9) return num;
        int ans = num%9;
        if(num!=0&&ans==0) return 9;
        else return ans;
    }
}